docs/guide/snaps-exports.md

```diff
@@ -0,0 +1,73 @@
+# Exports
+
+::: tip Developer Preview Software
+Snaps is pre-release software. To try Snaps, install [MetaMask Flask](https://metamask.io/flask).
+:::
+
+::: tip Feature Requests
+Do you have feature requests? Other ideas? We'd love to hear about them! [Click here](https://github.com/MetaMask/snaps-skunkworks/discussions) to join the discussion.
+:::
+
+## Table of Contents
+
+[[toc]]
+
+## `onRpcRequest`
+
+In order to communicate with the outside world, the snap must implement its own JSON-RPC API by exposing an exported function called `onRpcRequest`. Whenever the snap receives a JSON-RPC request from an external entity (a dapp or even another snap), this handler function will be called with the above parameters.
+
+::: tip Does my snap need to have an RPC API?
+Well, no, that's also up to you! If your snap can do something useful without receiving and responding to JSON-RPC requests, then you can skip exporting onRpcRequest. However, if you want to do something like manage the user's keys for a particular protocol and create a dapp that e.g. sends transactions for that protocol via your snap, you need to specify an RPC API.
+:::
+
+### Parameters
+
+- `RpcHandlerArgs` - The origin and the JSON-RPC request.
+
+```typescript
+interface RpcHandlerArgs = {
+  origin: string;
+  request: JsonRpcRequest<unknown[] | { [key: string]: unknown }>;
+};
+```
+
+### Returns
+
+```typescript
+type RpcHandlerReturn = Promise<unknown>;
+```
+
+`RpcHandlerReturn` - A promise containing the return of the implemented RPC Method.
+
+## `onTransaction`
+
+If the snap whats to provide transactions insights before they are accepted for signing by the uses, the snap must export a function called `onTransaction`. Whenever a transaction is submited to the extension this function will be called. The raw transaction payload, before it is signed, will be passed to this method.
+
+::: tip Requesting the transaction insight permission
+In order for the extension to call the `onTransaction` method of the snap, the `endowment:transaction-insight` permission must be requested. see [Permissions](./snaps-permissions.html#endowment-transaction-insight)
+:::
+
+### Parameters
+
+- `onTransactionArgs` - the raw transaction payload and the [CAIP2](https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-2.md) chain ID.
+
+```typescript
+interface OnTransactionArgs {
+  transaction: Record<string, unknown>;
```

Comment:
We don't explain the transaction anywhere in the documentation. We could at least direct people to [SIP-3](https://metamask.github.io/SIPs/SIPS/sip-3) apendix.